gluten-core/src/main/scala/io/glutenproject/extension/ColumnarOverrides.scala

```diff
@@ -545,19 +544,25 @@ case class TransformPreOverrides(isAdaptiveContextOrTopParentExchange: Boolean)
 // This rule will try to convert the row-to-columnar and columnar-to-row
 // into columnar implementations.
 case class TransformPostOverrides(session: SparkSession, isAdaptiveContext: Boolean)
-    extends Rule[SparkPlan] {
+    extends Rule[SparkPlan] with Logging {
   val columnarConf = GlutenConfig.getConf
   @transient private val planChangeLogger = new PlanChangeLogger[SparkPlan]()
 
   def replaceWithTransformerPlan(plan: SparkPlan): SparkPlan = plan match {
     case plan: RowToColumnarExec =>
-      val child = replaceWithTransformerPlan(plan.child)
-      logDebug(s"ColumnarPostOverrides RowToArrowColumnarExec(${child.getClass})")
-      BackendsApiManager.getSparkPlanExecApiInstance.genRowToColumnarExec(child)
-    // The ColumnarShuffleExchangeExec node may be the top node, so we cannot remove it.
-    // e.g. select /* REPARTITION */ from testData, and the AQE create shuffle stage will check
-    // if the transformed is instance of ShuffleExchangeLike, so we need to remove it in AQE mode
-    // have tested gluten-it TPCH when AQE OFF
+      val childTransformer = BackendsApiManager.getSparkPlanExecApiInstance
+        .genHiveTableScanExecTransformer(plan.child)
```

Comment:
Why not transform hiveExec in `TransformPreOverrides`, like ` case plan: HiveTableScanExec`.

Reply:
I moved this to `TransformPreOverrides`, and as HiveTableScanExec is a private class, so it can not be used like `case plan: HiveTableScanExec` , but use reflect instead